lib/ansible/modules/remote_management/dellemc/ome/ome_device_info.py

```diff
@@ -0,0 +1,418 @@
+                 type: list
+            device_service_tag:
+                 description:
+                    - A list of service tags are applicable for C(detailed_inventory)
+                      and C(subsystem_health).
+                 type: list
+            inventory_type:
+                description:
+                    - For C(detailed_inventory), it returns details of the specified inventory type.
+                type: str
+            filter:
+                description:
+                    - For C(basic_inventory), it filters the collection of devices.
+                      I(filter) query format should be aligned with OData standards.
+                type: str
+
+requirements:
+    - "python >= 2.7.5"
+author: "Sajna Shetty(@Sajna-Shetty)"
+'''
+
+EXAMPLES = """
+---
+- name: Retrieve basic inventory of all devices.
+  ome_device_info:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+
+- name: Retrieve basic inventory for devices identified by IDs 33333 or 11111 using filtering.
+  ome_device_info:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "basic_inventory"
+    system_query_options:
+      filter: "Id eq 33333 or Id eq 11111"
+
+- name: Retrieve inventory details of specified devices identified by IDs 11111 and 22222.
+  ome_device_info:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "detailed_inventory"
+    system_query_options:
+      device_id:
+        - 11111
+        - 22222
+
+- name: Retrieve inventory details of specified devices identified by service tags MXL1234 and MXL4567.
+  ome_device_info:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "detailed_inventory"
+    system_query_options:
+      device_service_tag:
+        - MXL1234
+        - MXL4567
+
+- name: Retrieve details of specified inventory type of specified devices identified by ID and service tags.
+  ome_device_info:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "detailed_inventory"
+    system_query_options:
+      device_id:
+        - 11111
+      device_service_tag:
+        - MXL1234
+        - MXL4567
+      inventory_type: "serverDeviceCards"
+
+- name: Retrieve subsystem health of specified devices identified by service tags.
+  ome_device_info:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "subsystem_health"
+    system_query_options:
+      device_service_tag:
+        - MXL1234
+        - MXL4567
+
+"""
+
+RETURN = """
+---
+msg:
+  type: str
+  description: Over all device information status.
+  returned: on error
+  sample: "Failed to fetch the device information"
+device_info:
+  type: dict
+  description: Returns the information collected from the Device.
+  returned: success
+  sample: {
```

Comment:
```suggestion
  sample: |
    {
```
That way the YAML parser should treat the whole JSON block correctly as a string.

Reply:
It still complains about invalid YAML at this position:
```
01:14 ERROR: Found 1 validate-modules issue(s) which need to be resolved:
01:14 ERROR: lib/ansible/modules/remote_management/dellemc/ome/ome_device_info.py:168:9: E313 RETURN is not valid YAML (100%)
```
https://app.shippable.com/github/ansible/ansible/runs/118514/1/console